pytest_tests/helpers/cluster.py

```diff
@@ -0,0 +1,324 @@
+from typing import Any
+
+import data_formatters
+from neofs_testlib.blockchain import RPCClient
+from neofs_testlib.hosting import Host, Hosting
+from neofs_testlib.hosting.config import ServiceConfig
+from test_control import wait_for_success
+
+
+@dataclass
+class NodeBase:
+    """
+    Represents a node of some underlying service
+    """
+
+    id: str
+    name: str
+    host: Host
+
+    def __init__(self, id, name, host) -> None:
+        self.id = id
+        self.name = name
+        self.host = host
+        self.construct()
+
+    def construct(self):
+        pass
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+    def __hash__(self):
+        return id(self.name)
+
+    def __str__(self):
+        return self.label
+
+    def __repr__(self) -> str:
+        return self.label
+
+    @property
+    def label(self) -> str:
+        return self.name
+
+    @wait_for_success(60, 1)
+    def start_service(self):
+        self.host.start_service(self.name)
+
+    @wait_for_success(60, 1)
+    def stop_service(self):
+        self.host.stop_service(self.name)
+
+    def get_wallet_password(self) -> str:
+        return self._get_attribute(_ConfigAttributes.WALLET_PASSWORD)
+
+    def get_wallet_path(self) -> str:
+        return self._get_attribute(
+            _ConfigAttributes.LOCAL_WALLET_PATH,
+            _ConfigAttributes.WALLET_PATH,
+        )
+
+    def get_wallet_config_path(self):
+        return self._get_attribute(
+            _ConfigAttributes.LOCAL_WALLET_CONFIG,
+            _ConfigAttributes.WALLET_CONFIG,
+        )
+
+    def get_wallet_public_key(self):
+        storage_wallet_path = self.get_wallet_path()
+        storage_wallet_pass = self.get_wallet_password()
+        return data_formatters.get_wallet_public_key(storage_wallet_path, storage_wallet_pass)
+
+    def _get_attribute(self, attribute_name: str, default_attribute_name: str = None) -> list[str]:
+        config = self.host.get_service_config(self.name)
+        if default_attribute_name:
+            return config.attributes.get(
+                attribute_name, config.attributes.get(default_attribute_name)
+            )
+        else:
+            return config.attributes.get(attribute_name)
+
+    def _get_service_config(self) -> ServiceConfig:
+        return self.host.get_service_config(self.name)
+
+
+class InnerRingNode(NodeBase):
+    """
+    Class represents inner ring node in a cluster
+
+    Inner ring node is not always the same as physical host (or physical node, if you will):
+        It can be service running in a container or on physical host
+    For testing perspective, it's not relevant how it is actually running,
+        since neofs network will still treat it as "node"
+    """
+
+    pass
+
+
+class S3Gate(NodeBase):
+    """
+    Class represents S3 gateway in a cluster
+    """
+
+    def get_endpoint(self) -> str:
+        return self._get_attribute(_ConfigAttributes.ENDPOINT)
```

Comment:
we can have a set of endpoins for gate.

Reply:
at first we need update testsetup to have multiple addresses in hosting